controllers/component_build_controller_pipeline.go

```diff
@@ -447,6 +447,7 @@ func generateCelExpressionForPipeline(component *appstudiov1alpha1.Component, gi
 	eventCondition := fmt.Sprintf(`event == "%s"`, eventType)
 
 	targetBranchCondition := fmt.Sprintf(`target_branch == "%s"`, targetBranch)
+	repoUrl := strings.TrimSuffix(strings.TrimSuffix(component.Spec.Source.GitSource.URL, ".git"), "/")
```

Comment:
We definitely need a separate function for it, like `getGitRepoUrl(component)`.

Reply:
yep